src/views/Moderator/ban_functions.tsx

```diff
@@ -29,13 +29,18 @@ function moderate(player_id, prompt, obj) {
                 input: "text",
                 showCancelButton: true,
             })
-            .then((reason) => {
-                obj.moderation_note = reason;
-                console.log(obj);
-                put("players/" + player_id + "/moderate", obj)
-                    .then(resolve)
-                    .catch(reject);
-            }, reject);
+            .then(({ value: reason }) => {
+                if (reason) {
```

Comment:
What behavior do we want here if `reason === ""`?

Previously, it would still send a request to the server and the server would respond with a Bad Request message.  Now the box closes with no error.

<img width="559" alt="Screen Shot 2022-06-19 at 4 31 44 PM" src="https://user-images.githubusercontent.com/25233703/174504378-fa9a030f-fa2c-4306-b672-133b0a186f7e.png">

Reply:
As discussed, I put in isConfirmed handlers where appropriate for all `input: "text"` alerts.

As I was typing a description of what I did, I realised I could do better, hence the second commit.

Sweetalert's input validation helps here.

Where before it made sense to defer validation to the server (for example, and empty moderation note might be OK, who knows?) it does that still.

Where actually some validation of the input text is obviously needed, I've done that.